### PR TITLE
Platops 1621 bcprov jdk15on upgrade

### PR DESCRIPTION
**Pull request by jm2dev**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -6,25 +6,25 @@ val appName = "http-caching-client"
 
 val compileDependencies = Seq(
   "com.typesafe.play" %% "play-json"       % "2.5.15" % "provided",
```
Comment (dabd) [inline build.sbt:8]: We can use play-json 2.5.19 here.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+help: ## Print this help
```
Comment (dabd) [inline Makefile]: What are the advantages of the Makefile?

File: .scalafmt.conf [hunk 3]
```diff
@@ -0,0 +1,45 @@
+style = defaultWithAlign
```
Comment (dabd) [inline .scalafmt.conf:1]: We usually make a commit with scalafmt config and the project after being formatted.  Then the PR is based on the already formatted project so it is easier to tell the code changes from formatting changes.

File: build.sbt [hunk 4]
```diff
@@ -6,25 +6,25 @@ val appName = "http-caching-client"
...
-  "uk.gov.hmrc"       %% "json-encryption" % "3.2.0",
+  "uk.gov.hmrc"       %% "json-encryption" % "4.0.0",
   "uk.gov.hmrc"       %% "http-core"       % "0.5.0"
```
Comment (dabd) [inline build.sbt:10]: We can get rid of http-core and use http-verbs instead

File: Makefile (hunk 2 above)
Comment (jm2dev) [reply Makefile]: I use a lot to provide simple tasks independently of the build tool:
```
make build
``` 
builds the project and do all the magic stuff I don't to remember per project, and it's the only command I need to remember. It provides a simple contract.
I can delete it if you prefer.

File: .scalafmt.conf (hunk 3 above)
Comment (jm2dev) [reply .scalafmt.conf:1]: I didn't know it, sorry.

File: build.sbt (hunk 4 above)
Comment (jm2dev) [reply build.sbt:10]: Done.

File: build.sbt (hunk 1 above)
Comment (jm2dev) [reply build.sbt:8]: Done.

**Pull request merged**